### PR TITLE
Make sure the appmap django middleware is always present

### LINKED ISSUE #141: Only one http_server_request captured for saleor
When mapping the tests in [saleor](https://github.com/mirumee/saleor), only the first HTTP request generates an `http_server_request` event. As it happens, the first test that makes an HTTP request is this one https://github.com/mirumee/saleor/blob/master/saleor/core/tests/test_middleware.py, which removes `appmap.django.Middleware` from`django.conf.settings.MIDDLEWARE`.

### PR DESCRIPTION
**Pull request by apotterri**
Make sure that our django middleware is always present in the stack, even if a test resets it.

Also, factor the django test app out from test_django.py into a separate app.

Fixes #141.

### CODE REVIEW

File: appmap/django.py [hunk 1]
```diff
@@ -288,7 +288,6 @@ def inject_middleware():
 def load_middleware(*args, **kwargs):
     """Patched wrapper to inject AppMap middleware first"""
     inject_middleware()
-    BaseHandler.load_middleware = original_load_middleware
```
Comment (apotterri) [inline appmap/django.py:291]: @divide resetting to the original `load_middleware` function here causes problems if the test sets the middleware (as happens here https://github.com/mirumee/saleor/blob/master/saleor/core/tests/test_middleware.py#L16).

Did you do this just for the sake of efficiency, or was there some other reason for it?

**Review by torchello — APPROVED** [review]
Looks ok!

**Comment by kgilpin** [comment]
Is this done?

**Comment by apotterri** [comment]
Yup.

**Pull request merged**